validation/rc-exp2syn-spike/neuron-rc-exp2syn-spike.py

```diff
@@ -104,26 +113,56 @@ def f():
             rec[idx] = h.t
     return f
 
-for i in range(ncell):
+for i in range(len(nc_record)):
     nc_record[i].record(recorder(spike, i))
 
 h.dt = dt
 h.steps_per_ms = 1/dt # or else NEURON might noisily fudge dt
 h.secondorder = 2
 h.tstop = tend
-h.run()
 
-# Collect and save data
+if not coreneuron:
+    h.finitialize()
+    h.run()
 
-v0 = list(cell0_v)
-ts = list(sample_t)
-out = xarray.Dataset(
-    {'v0': (['time'], v0),
-     'spike': (['gid'], spike),
-     'delay': (['gid'], delays)
-    }, coords={'time': ts, 'gid': list(range(0,ncell))})
+    # Collect and save data
 
-for v in param_vars:
-    out[v] = np.float64(globals()[v])
+    v0 = list(cell0_v)
+    ts = list(sample_t)
+    out = xarray.Dataset(
+        {'v0': (['time'], v0),
+         'spike': (['gid'], spike),
+         'delay': (['gid'], delays)
+        }, coords={'time': ts, 'gid': list(range(0,ncell))})
+
+    for v in param_vars:
+        out[v] = np.float64(globals()[v])
+
+    out.to_netcdf(output)
+
+else:
+    h.cvode.cache_efficient(1)
+
+    pc.set_maxstep(mindelay)
+    h.finitialize()
+
+    cwd = os.getcwd()
+    with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
Can we have a quick comment that explains what is being done, and why, here?
The reason for this rather convoluted approach should be made clear (otherwise someone reading this code might think you are crazy)